src/ape/types/__init__.py

```diff
@@ -39,7 +41,13 @@
 cases.
 """
 
-AddressType = ChecksumAddress  # type: ignore
+DependencyType = Union["IPFSDependency", "GithubDependency"]
+"""
+A project dependency, such as a :class:`ape.managers.config.IPFSDependency`
+or a :class:`ape.managers.config.GithubDependency`.
+"""
+
+AddressType = NewType("AddressType", HexAddress)  # type: ignore
```

Comment:
I like using the `ChecksumAddress` assumption though, at least for processing user input, to ensure they didn't make a mistake typing the address

Reply:
`mypy` just really hates this lol I keep fighting it